PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelper.cxx

```diff
@@ -798,7 +883,29 @@ void AliAnalysisTaskEmcalEmbeddingHelper::DetermineFirstFileToEmbed()
   if (fFilenameIndex == -1 && fRandomFileAccess) {
     // Floor ensures that we it doesn't overflow
     TRandom3 rand(0);
-    fFilenameIndex = TMath::FloorNint(rand.Rndm()*fFilenames.size());
+
+    // In case of a fixed block run range
+    // make first one of the files within the run range
+    if ( fEmbeddedRunblock.size() != 0 )
+    {
+      UInt_t iter = 0;
+      UInt_t nfiles = fFilenames.size();
+      Bool_t ok = kFALSE;
+      // Iterate but not too much
+      while ( !ok && iter <= nfiles*10 )
+      {
+        iter++;
+        fFilenameIndex = TMath::FloorNint(rand.Rndm()*fFilenames.size());
+        const char* path = (fFilenames.at(fFilenameIndex)).c_str();
+        ok = IsRunInRunblock(path);
+      }
+      
+      if ( !ok ) 
+        AliFatal("No file found with the run range requeirements, check list of MC files and block run ranges STOP!");
+    }
```

Comment:
If you remove the filenames that are not of interest, I think you don't need to modify this function at all

Reply:
But this is what is done here, remove the filenames not of interest. I have not seen where to do this before here.